crates/nargo_cli/src/manifest.rs

```diff
@@ -131,19 +159,27 @@ enum DependencyConfig {
 
 impl DependencyConfig {
     fn resolve_to_dependency(&self, pkg_root: &Path) -> Result<Dependency, ManifestError> {
-        match self {
+        let dep = match self {
             Self::Github { git, tag } => {
                 let dir_path = clone_git_repo(git, tag).map_err(ManifestError::GitError)?;
                 let toml_path = dir_path.join("Nargo.toml");
                 let package = resolve_package_from_toml(&toml_path)?;
-                Ok(Dependency::Remote { package })
+                Dependency::Remote { package }
             }
             Self::Path { path } => {
                 let dir_path = pkg_root.join(path);
                 let toml_path = dir_path.join("Nargo.toml");
                 let package = resolve_package_from_toml(&toml_path)?;
-                Ok(Dependency::Local { package })
+                Dependency::Local { package }
             }
+        };
+
+        // Cannot depend on a binary
+        // TODO: Can we depend upon contracts?
```

Comment:
I think so. Need to look at how we handle cross-contract calls atm but that must require importing the other contract, no?

Reply:
This will need further research